### PR TITLE
Improved argument parsing

### PR DESCRIPTION
**Pull request by aspeake**
Split out ecm_prep.py and run.py argument handling into separate functions to enable easier testing and improve organization/readability.

Summary of updates:
- **ecm_prep.py**: new imported file `ecm_prep_args.py` to handle argument definitions, parsing, and user input requests. 
- **ecm_prep.py**: new optional argument `alt_regions_option` to circumvent the need for a user response to specify EMM, AIA or State breakdowns. 
- **run.py**: avoid unnecessary global variable `opts`, instead directly pass this to `main()` and `Engine.__init__()`
- **run.py tests**: updates required to match new `Engine()` constructor params

These updates enable easier access to the `main()` functions from other scripts and provide user inputs, making it easier to implement integration testing, profiling, and unit tests for argument parsing. Functionality is unchanged; PRs that aim to organize the project into modules and/or subpackages may alter this methods further in the future.

### CODE REVIEW

File: ecm_prep.py [hunk 1]
```diff
@@ -13742,94 +13360,10 @@ def main(base_dir):
...
-                              "breakouts as applicable"))
-    # Optional flag to include accounting of fugitive emissions from
-    # refrigerant leakage and/or supply chain methane leakage
-    parser.add_argument("--fugitive_emissions", action="store_true",
-                        help="Account for fugitive emissions sources")
-    # Object to store all user-specified execution arguments
-    opts = parser.parse_args()
+    opts = ecm_args(sys.argv[1:])
 
     # Set current working directory
```
Comment (trynthink) [inline ecm_prep.py:13365]: Please delete this comment line or move it to before line 12574, where the getcwd() call is now located?

File: ecm_prep_args.py [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+    parser.add_argument("--fugitive_emissions", action="store_true",
+                        help="Account for fugitive emissions sources")
+    # Object to store all user-specified execution arguments
+    opts = parser.parse_args(args)
+    opts = fill_user_inputs(opts)
+
+    return opts
+
+
+def fill_user_inputs(opts: argparse.NameSpace) -> argparse.NameSpace:  # noqa: F821
```
Comment (trynthink) [inline ecm_prep_args.py:105]: Please add a brief Google-style docstring to this function that explains what the function does plus args and returns.

File: run_test.py [hunk 3]
```diff
@@ -26905,8 +26896,12 @@ def test_sbmkt_frac_add(self):
...
     """Trigger default behavior of running all test fixtures in the file."""
-    unittest.main()
+    # runner = unittest.TextTestRunner()
+    # itersuite = unittest.TestLoader().loadTestsFromTestCase(MyTestClass)
+    # runner.run(itersuite)
+    # # unittest.main(argv=[''])
 
 
 if __name__ == "__main__":
+
```
Comment (trynthink) [inline run_test.py:26906]: Did this newline get added inadvertently? I don't think it's needed.

File: run_test.py [hunk 4]
```diff
@@ -16164,6 +16155,8 @@ class ComCompeteTest(unittest.TestCase, CommonMethods):
 
     @classmethod
     def setUpClass(cls):
+
+        print("$#@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@")
```
Comment (trynthink) [inline run_test.py:16159]: What is this line doing?

Comment (aspeake) [reply run_test.py]: Nothing, deleted

**Review by trynthink — APPROVED** [review]

**Pull request merged**